src/django/pfb_analysis/models.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import unicode_literals
+
+import os
+
+from django.db import models
+
+from localflavor.us.models import USStateField
+import us
+
+from pfb_network_connectivity.models import PFBModel
+
+
+def get_neighborhood_file_upload_path(instance, filename):
+    """ Upload each boundary file to its own directory """
+    return 'neighborhood_boundaries/{0}/{1}'.format(instance.name, os.path.basename(filename))
+
+
+class Neighborhood(models.Model):
+    """Neighborhood boundary used for an AnalysisJob """
+
+    def __repr__(self):
+        return "<Neighborhood: {}, ({})>".format(self.name, self.label)
+
+    name = models.SlugField(max_length=256, primary_key=True)
+    label = models.CharField(max_length=256, help_text='Human-readable label for boundary')
+    state_abbrev = USStateField(help_text='The US state that the uploaded neighborhood is in')
```

Comment:
Not directly with model field options at the database level, because of the way Django handles CharFields. I updated the model's save method to call `full_clean()` which forces validation before save. This doesn't happen by default in django for reasons.